frontend/src/data/variables/VariableProvider.ts

```diff
@@ -14,17 +18,17 @@ abstract class VariableProvider {
     this.providesMetrics = providesMetrics;
   }
 
-  async getData(breakdowns: Breakdowns): Promise<MetricQueryResponse> {
-    if (!this.allowsBreakdowns(breakdowns)) {
+  async getData(metricQuery: MetricQuery): Promise<MetricQueryResponse> {
+    if (!this.allowsBreakdowns(metricQuery.breakdowns)) {
       return createMissingDataResponse(
         "Breakdowns not supported for provider " +
           this.providerId +
           ": " +
-          breakdowns.getUniqueKey()
+          metricQuery.breakdowns.getUniqueKey()
       );
     }
```

Comment:
Now that this accepts a MetricQuery, I think it would be nice to throw an error if the query contains a metric that is not in `this.providesMetrics`. I think it should be a normal error rather than a missing data response, because that would indicate a bug in our variable mapping logic